tests/linearGaussian_snpe_test.py

```diff
@@ -571,6 +571,7 @@ def simulator(theta):
         condition=samples[0],
         dims_to_sample=[dim_to_sample_1, dim_to_sample_2],
     )
+    conditioned_potential_fn.set_x(x_o, x_is_iid=False)
```

Comment:
Hm why is this needed now? Can we not use `x_o` from the `potential_fn` that was used to generate the `conditioned_potential_fn`?